### PR TITLE
Input ORC format

### PR DESCRIPTION
**Pull request by akonyaev90**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Category (leave one):
- New Feature

Short description (up to few sentences):
Add input ORC format
...

For read data in ORC format.
Made similar to parquet.
The file we are reading should fit in memory.
...
https://github.com/yandex/ClickHouse/issues/4903

### CODE REVIEW

File: contrib/arrow-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -47,6 +47,61 @@ target_include_directories(${THRIFT_LIBRARY} SYSTEM PUBLIC ${ClickHouse_SOURCE_D
...
+set(ORC_BUILD_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/../orc/c++/include)
+
+set(GOOGLE_PROTIBUF_DIR ${ClickHouse_SOURCE_DIR}/contrib/protobuf/src/)
+set(ORC_ADDITION_SOURCE_DIR ${ClickHouse_SOURCE_DIR}/contrib/arrow-cmake/cpp/src/orc/)
+set(ARROW_SRC_DIR ${ClickHouse_SOURCE_DIR}/contrib/arrow/cpp/src)
+
+include_directories(SYSTEM ${ORC_INCLUDE_DIR})
+include_directories(SYSTEM ${ORC_SOURCE_SRC_DIR})
+include_directories(SYSTEM ${ORC_SOURCE_WRAP_DIR})
+include_directories(SYSTEM ${GOOGLE_PROTIBUF_DIR})
```
Comment (proller) [inline contrib/arrow-cmake/CMakeLists.txt]: PROTIBUF ?

File: dbms/src/Processors/Formats/Impl/ORCBlockInputFormat.cpp [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+    Chunk res;
+    Columns columns_list;
+    UInt64 num_rows = 0;
+    auto & header = getPort().getHeader();
+    columns_list.reserve(header.rows());
+
+    if (!in.eof())
+    {
+        if (row_group_current < row_group_total)
+            throw Exception{"Got new data, but data from previous chunks not readed " + std::to_string(row_group_current) + "/" + std::to_string(row_group_total), ErrorCodes::CANNOT_READ_ALL_DATA};
```
Comment (alexey-milovidov) [inline dbms/src/Processors/Formats/Impl/ORCBlockInputFormat.cpp]: not readed -> was not read

**Comment by alexey-milovidov** [comment]
Is it possible to generate protobuf on the fly? (auto-generated file does not look good).

File: contrib/CMakeLists.txt [hunk 3]
```diff
@@ -10,6 +10,13 @@ endif ()
 
 set_property(DIRECTORY PROPERTY EXCLUDE_FROM_ALL 1)
 
+#===== add orc subdir
```
Comment (proller) [inline contrib/CMakeLists.txt]: if (USE_INTERNAL_ORC_LIBRARY) ...

**Comment by akonyaev90** [comment]
Is it possible to generate protobuf on the fly? - Alexey, what file are you talking about?

**Comment by akonyaev90** [comment]
orc_proto.pb.cc ?

**Comment by akonyaev90** [comment]
now orc_proto building with proto-command (I added command in arrow-cmake cmakeFile)
typos corrected

File: cmake/find_orc.cmake [hunk 4]
```diff
@@ -0,0 +1,8 @@
+##TODO replace hardcode to find procedure
+
```
Comment (abyss7) [inline cmake/find_orc.cmake]: @akonyaev90 when are you going to implement the proper build logic for ORC contrib?

**Comment by guanlisheng** [comment]
may I know when this will be released?

**Pull request merged**